### PR TITLE
Modified code so list of Metadatas is not deep copied when cloning Fe…

### PR DESCRIPTION
**Pull request by hbs**
…tchRequest

### CODE REVIEW

**Comment by ftence** [comment]
The token is also deep copied, should it be avoided too? In that case, a utility function clearly stating it is shallow copy would be nicer.

**Comment by hbs** [comment]
The token is not a heavy object to copy, the list of metadata can contain hundreds of thousands of instances, making it a real hit in terms of performance and memory consumption

File: warp10/src/main/java/io/warp10/continuum/store/ParallelGTSDecoderIteratorWrapper.java [hunk 1]
```diff
@@ -301,11 +301,21 @@ public ParallelGTSDecoderIteratorWrapper(FetchRequest req, boolean optimized, Ke
         GTSDecoderIterator iterator = null;
         
         if (optimized) {
+          // Remove Metadatas from FetchRequest otherwise new FetchRequest(req) will do a deep copy
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/continuum/store/ParallelGTSDecoderIteratorWrapper.java:304]: Factorize code outside conditional.

File: warp10/src/main/java/io/warp10/continuum/store/ParallelGTSDecoderIteratorWrapper.java [hunk 2]
```diff
@@ -320,11 +330,21 @@ public ParallelGTSDecoderIteratorWrapper(FetchRequest req, boolean optimized, Ke
       GTSDecoderIterator iterator = null;
       
       if (optimized) {
+        // Remove Metadatas from FetchRequest otherwise new FetchRequest(req) will do a deep copy
```
Comment (ftence) [inline warp10/src/main/java/io/warp10/continuum/store/ParallelGTSDecoderIteratorWrapper.java:333]: Also factorize here.

File: warp10/src/main/java/io/warp10/continuum/store/OptimizedSlicedRowFilterGTSDecoderIterator.java [hunk 3]
```diff
@@ -193,7 +197,11 @@ public boolean hasNext() {
       if (null != iterator) {
         try { iterator.close(); } catch (Exception e) {}
       }
+      // Remove Metadatas from FetchRequest otherwise new FetchRequest(req) will do a deep copy
```
[the remark this answers is not in the document]
Comment (hbs) [reply warp10/src/main/java/io/warp10/continuum/store/OptimizedSlicedRowFilterGTSDecoderIterator.java]: Indeed, Metadatas is unset

**Comment by hbs** [comment]
PTAL

**Pull request merged**